src/components/NFTPage/Recommendation.vue

```diff
@@ -88,60 +94,139 @@ import { Swiper, SwiperSlide } from 'vue-awesome-swiper';
 
 export default {
   name: 'NFTPageRecommendation',
   components: {
     Swiper,
     SwiperSlide,
   },
-  mixins: [bookstoreMixin],
+  mixins: [bookstoreMixin, nftOrCollection, walletMixin],
   props: {
-    iscnOwner: {
+    classId: {
+      type: String,
+      default: '',
+    },
+    collectionId: {
       type: String,
-      default: undefined,
+      default: '',
     },
-    isFollowed: {
+    // if true, show author works, else show default featured works
+    isAuthorSpecific: {
       type: Boolean,
-      default: false,
+      default: true,
     },
-    recommendedList: {
-      type: Array,
-      default: () => [],
+  },
+  computed: {
+    ...mapGetters(['walletHasLoggedIn', 'walletFollowees']),
+    type() {
+      if (this.nftIsWritingNFT) {
+        return 'writingNFT';
+      }
+      if (this.classId) {
+        return 'book';
+      }
+      return 'collection';
     },
-    isLoading: {
-      type: Boolean,
-      default: false,
+    isFollowed() {
+      return this.walletFollowees?.includes(this.productOwner) || false;
     },
-    shouldShowFollowButton: {
-      type: Boolean,
-      default: false,
+    shouldShowFollowButton() {
+      return (
+        this.wallet !== this.getAddress &&
+        this.walletHasLoggedIn &&
+        !this.isFollowed
+      );
     },
-    isBookNft: {
-      type: Boolean,
-      default: false,
+    hasPublisher() {
+      if (this.type === 'collection') {
+        return true;
+      }
+      return !!this.iscnWorkAuthor;
     },
-    displayItemCount: {
-      type: [Number, String],
-      default: 5,
+    defaultFeaturedWNFT() {
+      return DEFAULT_RECOMMENDATIONS_LIST.WNFT.map(nft => ({
+        classId: nft,
+      }));
     },
-    shouldShowIscnOwner: {
-      type: Boolean,
-      default: true,
+    defaultFeaturedBooks() {
+      return this.bookstoreListItemForLandingPage?.length
+        ? this.bookstoreListItemForLandingPage
+        : DEFAULT_RECOMMENDATIONS_LIST.BOOK.map(nft => ({ classId: nft }));
     },
-    iscnWorkAuthor: {
-      type: String,
-      default: undefined,
+    defaultFeaturedList() {
+      if (this.nftIsWritingNFT) {
+        return this.defaultFeaturedWNFT;
+      }
+      return this.defaultFeaturedBooks;
+    },
+    recommendedList() {
+      if (!this.isAuthorSpecific) {
+        return this.defaultFeaturedBooks;
+      }
+      const createdList =
+        this.getCreatedNFTClassesByAddress(this.productOwner) || [];
+      const createdClassIds = new Set(createdList.map(item => item.classId));
+
+      let recommendedList = [
+        ...createdList,
+        ...this.defaultFeaturedList.filter(
+          item =>
+            item.classId !== this.classId && !createdClassIds.has(item.classId)
+        ),
+      ];
+      const featuredSet =
+        this.getNFTClassFeaturedSetByAddress(this.iscnOwner) || new Set();
+      const hiddenSet =
+        this.getNFTClassHiddenSetByAddress(this.iscnOwner) || new Set();
```

Comment:
When do we use `productOwner` vs `iscnOwner`?

Reply:
[🎨 Use computed property for type & owner determination](https://github.com/likecoin/liker-land/pull/1863/commits/96ef0db4fb6b5a48870f8ec7cada2148278e5902)